twitter_api/main.py

```diff
@@ -16,6 +21,158 @@ def before_request():
+                          [profile['user_id']])
+        for t in tweets:
+            t['uri'] = '/tweet/{}'.format(t['id'])
+            created_date = datetime.strptime(t['date'], '%Y-%m-%d %H:%M:%S')
+            t['date'] = created_date.strftime("%Y-%m-%dT%H:%M:%S")
+        profile['tweet_count'] = len(tweets)
+        profile['tweets'] = tweets
+
+        return Response(response=json.dumps(profile), status=200, mimetype="application/json")
+
+    return abort(404)
+
+
+@app.route('/profile', methods=['POST'])
+@auth_only
+@json_only
+def post_profile():
+    required_attrs = ['first_name', 'last_name', 'birth_date']
+    data = request.get_json()
+    user_id = get_user_id_from_token(data['access_token'])
+    if not user_id:
+        abort(401)
+    if all(attr in data for attr in required_attrs):
+        query_db('update user set first_name=?, last_name=?, birth_date=? where id=?',
+                 (data['first_name'], data['last_name'],
+                  data['birth_date'],
+                  int(user_id)))
+        g.db.commit()
+        return Response(response='', status=201)
+    abort(400)
+
+
+@app.route('/tweet/<tweet_id>')
+def get_tweet(tweet_id):
+    tweet = query_db('select t.id, t.content, t.created as date, u.username as profile '
+                     'from tweet t inner join user u on t.user_id = u.id '
+                     'where t.id=?', [tweet_id], one=True)
+    if tweet:
+        created_date = datetime.strptime(tweet['date'], '%Y-%m-%d %H:%M:%S')
+        tweet['date'] = created_date.strftime("%Y-%m-%dT%H:%M:%S")
+        tweet['profile'] = '/profile/{}'.format(tweet['profile'])
+        tweet['uri'] = '/tweet/{}'.format(tweet['id'])
+
+        return Response(response=json.dumps(tweet), status=200, mimetype='application/json')
+
+    abort(404)
+
+
+@app.route('/tweet', methods=['POST'])
+@auth_only
+@json_only
+def post_tweet():
+    required_attrs = ['content']
+    data = request.get_json()
+    user_id = get_user_id_from_token(data['access_token'])
+    if not user_id:
+        abort(401)
+    if all(attr in data for attr in required_attrs):
+        query_db('insert into tweet (user_id, content) values(?, ?)',
+                 (user_id, data['content']))
+        g.db.commit()
+        return Response(response='', status=201)
+    abort(400)
+
+
+@app.route('/tweet/<tweet_id>', methods=['DELETE'])
+@auth_only
+@json_only
+def delete_tweet(tweet_id):
+    data = request.get_json()
+    tweet = query_db('select id, user_id '
+                     'from tweet '
+                     'where id=?', [tweet_id], one=True)
+    if tweet:
+        if tweet['user_id'] == get_user_id_from_token(data['access_token']):
+            query_db('delete from tweet where id=?', [tweet_id])
+            g.db.commit()
+
+            return Response(response='', status=204)
+        else:
+            # tweet does not belong to user
+            abort(401)
+
+    abort(404)
```

Comment:
(applies for other functions too since they are constructed similarly)
imho it might be better to do
```
if tweet:
  # existing code
else:
  abort(404)
```

reason is because right now, it looks like the default is to abort. which is rather strange. connecting it to the logic before using an `if/else` statement makes the flow a lot more obvious. also, since it's only one line, an extra indented line is okay, imho.